array.go

```diff
@@ -1198,6 +1198,254 @@ func (a *Array) NotContainsAny(values ...interface{}) *Array {
 	return a
 }
 
+// IsOrdered succeeds if array is ordered based on optional `comparator` function.
+// For default, it will use built-in comparator function for each data type.
+// Built-in comparator requires all elements in the array to have same data type.
+// Array with 0 or 1 element will always pass
+// Example:
+//
+//	array := NewArray(t, []interface{}{100, 101, 102})
+//	array.IsOrdered() // succeeds
+//	array.IsOrdered(func(x, y *httpexpect.Value) bool {
+//		return x.Number().Raw() < y.Number().Raw()
+//	}) // succeeds
+func (a *Array) IsOrdered(less ...func(x, y *Value) bool) *Array {
+	a.chain.enter("IsOrdered()")
+	defer a.chain.leave()
+
+	if a.chain.failed() {
+		return a
+	}
+
+	if len(less) > 1 {
+		a.chain.fail(AssertionFailure{
+			Type: AssertUsage,
+			Errors: []error{
+				errors.New("unexpected multiple less arguments"),
+			},
+		})
+		return a
+	}
+
+	if len(a.value) <= 1 {
+		return a
+	}
+
+	var lessFn func(x, y *Value) bool
+	if len(less) == 1 {
+		lessFn = less[0]
+	} else {
+		lessFn = a.getBuiltinComparator()
+		if a.chain.failed() {
+			return a
+		}
+	}
+
+	var xChainFailure, yChainFailure bool
+	for i := 0; i < int(len(a.value))-1; i++ {
+		xChain := a.chain.clone()
+		xChain.replace("IsOrdered[%d]", i)
+		xChain.setFailCallback(func() {
+			xChainFailure = true
+		})
+		x := newValue(xChain, a.value[i])
+
+		yChain := a.chain.clone()
+		yChain.replace("IsOrdered[%d]", i+1)
+		yChain.setFailCallback(func() {
+			yChainFailure = true
+		})
+		y := newValue(yChain, a.value[i+1])
+
+		if lessFn(y, x) {
+			a.chain.fail(AssertionFailure{
+				Type:      AssertLt,
+				Actual:    &AssertionValue{x.value},
+				Expected:  &AssertionValue{y.value},
+				Reference: &AssertionValue{a.value},
+				Errors: []error{
+					errors.New("expected: array is ordered"),
+					fmt.Errorf("element %v must be less than element %v, but it isn't",
+						x.value, y.value),
+				},
+			})
+			return a
+		}
+	}
+
+	if xChainFailure || yChainFailure {
+		a.chain.setFailed()
+	}
+
+	return a
+}
+
+// NotOrdered is opposite to `IsOrdered`.
```

Comment:
Given this another thought..

These functions are not fully opposite because of handling of arrays if 0 or 1 elements. 

So, maybe its better to have explicit comments to both and not state that they are opposite. 

For IsOrdered, let's say that it succeeds if every element is not less than the previous element.

For NotOrdered, let's say that it succeeds if at least one element is less than the previous element. 

For both, let's mention that for 0 and 1 elements we always succeed.

Reply:
done